test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+
+    let Some(lsb) = get_field("lsb") else {
+        let err_msg = "Missing ".to_string() + "reqid.lsb field";
+        error!("{}", err_msg);
+        return Err(serde::de::Error::custom(err_msg));
+    };
+
+    let Some(msb) = get_field("msb") else {
+        let err_msg = "Missing ".to_string() + "reqid.msb field";
+        error!("{}", err_msg);
+        return Err(serde::de::Error::custom(err_msg));
+    };
+
+    Ok(UUID {
+        lsb,
+        msb,
+        special_fields: SpecialFields::default(),
+    })
+}
+
+#[derive(Default)]
+pub struct WrapperUPayload(pub UPayload);
+impl<'de> Deserialize<'de> for WrapperUPayload {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+        let Ok(upayload) = parse_upayload(&value) else {
+            let err_msg = "Error parsing payload: ".to_string();
+            return Err(serde::de::Error::custom(err_msg));
+        };
+
+        Ok(WrapperUPayload(upayload))
+    }
+}
+
+#[allow(clippy::map_unwrap_or)]
+fn parse_upayload(value: &Value) -> Result<UPayload, serde_json::Error> {
+
+    let format = value
+        .get("format")
+        .and_then(|format_value| format_value.as_str())
+        .map(|format_str| {
+            UPayloadFormat::from_str(format_str).unwrap_or_else(|| {
+                error!("Error: Unable to parse string to UPayloadFormat");
+                UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED
+            })
+        })
+        .unwrap_or_else(|| {
+            error!("Error: value of format is not a string");
+            UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED
+        });
+
+    let length = value
+        .get("length")
+        .and_then(Value::as_str)
+        .and_then(|length_value| length_value.parse::<i32>().ok());
+
+    let data = if let Some(data_value) = value.get("value") {
+        if let Ok(data_vec) = serde_json::to_vec(data_value) {
+            Some(Data::Value(data_vec))
+        } else {
+            Some(Data::Reference(0))
+        }
+    } else {
+        return Err(serde::de::Error::custom("Missing value field"));
+    };
+
+    Ok(UPayload {
+        format: format.into(),
+        length,
+        data,
+        special_fields: SpecialFields::default(),
+    })
+}
+
+#[derive(Debug, Default)]
+
+pub struct WrapperUMessage(pub UMessage);
+
+impl<'de> Deserialize<'de> for WrapperUMessage {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+
+        let wattributes = match value.get("attributes") {
+            Some(attributes) => {
+                serde_json::from_value::<WrapperUAttribute>(attributes.clone()).unwrap_or_default()
+            }
+            None => WrapperUAttribute(UAttributes::default()),
+        };
+
+        let wpayload = match value.get("payload") {
```

Comment:
Same sort of comment as [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1597249312).

Reply:
tried my best on this, but implementation may be a bit rough, let's take a look